runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java

```diff
@@ -203,6 +203,9 @@ static StreamExecutionEnvironment createStreamExecutionEnvironment(
             .getCheckpointConfig()
             .setMinPauseBetweenCheckpoints(minPauseBetweenCheckpoints);
       }
+    } else {
+      // checkpointing is disabled, we can allow shutting down sources when they're done
```

Comment:
Is there a Flink JIRA for this?

Reply:
https://issues.apache.org/jira/browse/FLINK-2491